### PR TITLE
domain tests

### PR DESCRIPTION
**Pull request by wiewiurdp**

### CODE REVIEW

File: module/account/tests/Domain/ValueObject/EmailTest.php [hunk 1]
```diff
@@ -26,6 +26,7 @@ public function testValidaValue(string $value): void
         $email = new Email($value);
         $this->assertEquals($value, $email->getValue());
         $this->assertEquals($value, (string) $email);
+        $this->assertTrue($email->isEqual($email));
```
Comment (rprzedzik) [inline module/account/tests/Domain/ValueObject/EmailTest.php]: This Assertion have no sense in this test "testValidaValue". 
Tests the methods on its own, what about different objects of the same value or with different values ?
This should be on separate test.

File: module/attribute/tests/Domain/ValueObject/OptionValue/MultilingualOptionTest.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ */
+class MultilingualOptionTest extends TestCase
+{
+    /**
+     */
+    public function testValueCreation(): void
+    {
+        /** @var TranslatableString | MockObject $value */
```
Comment (rprzedzik) [inline module/attribute/tests/Domain/ValueObject/OptionValue/MultilingualOptionTest.php]: This is not a MockObject

File: module/category-tree/tests/Domain/ValueObject/NodeTest.php [hunk 3]
```diff
@@ -51,12 +51,17 @@ public function testSettingParent(): void
 
     /**
      */
-    public function testAddChildren():void
+    public function testAddChildren(): void
     {
+        $categoryId2 = new CategoryId('cde02652-70ce-484e-bc9d-3bf61391522d');
```
Comment (rprzedzik) [inline module/category-tree/tests/Domain/ValueObject/NodeTest.php]: Its betet use MockedObject

File: module/product/tests/Domain/ValueObject/SkuTest.php [hunk 4]
```diff
@@ -29,6 +29,14 @@ public function testGetValue(string $sku): void
         $this->assertEquals($sku, $sku->getValue());
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testInvalidVAlue(): void
+    {
+        $sku = new Sku('yKL3rWXluEM7NapnMSVpYHpicQJkpJ0Obfx91mma0xwnQxUfsrwy5Nfki1LUZR4qYolBTlUFDO4RkeINIkjPzMfTSit0bQZJevXA6GMsj0LnSZaiT1bBfr00vtKWqLAPollonRzb6GBVlT5U9I6nC49r3Vnj2jUgpA73CvfnVnFBNnHqCaI2Cu48SKaVSRGgROhoD1dGPvvq98okavZ3ktVHk0TcyyiyfoH52U3gP3J5bNTVZngivjPJAqtOW8TO');
```
Comment (rprzedzik) [inline module/product/tests/Domain/ValueObject/SkuTest.php]: This entry will be incompatible with the currently implemented formatting rules. Maximum 120 characters per line.

File: module/attribute/tests/Domain/ValueObject/OptionValue/StringOptionTest.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    public function testValueCreation(): void
+    {
+        $value = 'value';
+
+        $valueObject = new StringOption($value);
+
+        $this->assertSame($value, $valueObject->getValue());
+        $this->assertSame(StringOption::TYPE, $valueObject->getType());
+        $this->assertSame($value, (string) $valueObject);
+        $this->assertTrue($valueObject->equal($valueObject));
```
Comment (rprzedzik) [inline module/attribute/tests/Domain/ValueObject/OptionValue/StringOptionTest.php]: Value Object equality should be in separate test for all possibilities.

File: module/attribute/tests/Domain/ValueObject/OptionValue/MultilingualOptionTest.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        $value = new TranslatableString(['en' => 'english', 'pl' => 'polish']);
+
+        $valueObject = new MultilingualOption($value);
+
+        $this->assertSame($value, $valueObject->getValue());
+        $this->assertSame(MultilingualOption::TYPE, $valueObject->getType());
+        $this->assertSame('english,polish', (string) $valueObject);
+        $this->assertTrue($valueObject->isMultilingual());
+        $this->assertTrue($valueObject->equal($valueObject));
```
Comment (rprzedzik) [inline module/attribute/tests/Domain/ValueObject/OptionValue/MultilingualOptionTest.php]: Value Object equality should be in separate test for all possibilities.

File: module/attribute-unit/tests/Domain/ValueObject/UnitTest.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    public function testValueCreation(): void
+    {
+        $code = ' code ';
+        $check = 'CODE';
+
+        $valueObject = new Unit($code);
+
+        $this->assertSame($check, $valueObject->getCode());
+        $this->assertSame($check, (string) $valueObject);
+        $this->assertSame($check, Unit::fromString($code)->getCode());
```
Comment (rprzedzik) [inline module/attribute-unit/tests/Domain/ValueObject/UnitTest.php]: Should be in separate test

File: module/category-tree/tests/Domain/ValueObject/NodeTest.php (hunk 3 above)
Comment (wiewiurdp) [reply module/category-tree/tests/Domain/ValueObject/NodeTest.php]: When I use MockObject, don't know why the assertion on 'hasChildren' method fails.

**Pull request merged**